File: app/src/main/java/com/hypnoweb/hypnowebapp/catalog/repo/network/CatalogNetworkDataSource.java
package com.hypnoweb.hypnowebapp.catalog.repo.network;

import com.hypnoweb.hypnowebapp.catalog.repo.Catalog;

import javax.inject.Inject;

import io.reactivex.Single;

/**
 * Created by stefan on 12/9/2017.
 */

public class CatalogNetworkDataSource {

    private CatalogApi catalogApi;
    private CatalogNetworkMapper catalogNetworkMapper;

    @Inject
    public CatalogNetworkDataSource(CatalogApi catalogApi, CatalogNetworkMapper catalogNetworkMapper) {
        this.catalogApi = catalogApi;
        this.catalogNetworkMapper = catalogNetworkMapper;
    }


    public Single<Catalog> getCatalog() {
        return catalogApi.getCatalog()
                .flatMap(catalogNetwork -> catalogNetworkMapper.fromNetworkCatalog(catalogNetwork));
    }
}

File: app/src/main/java/com/hypnoweb/hypnowebapp/catalog/repo/network/CatalogNetworkDataSource.kt
package com.hypnoweb.hypnowebapp.catalog.repo.network


import android.content.Context
import com.google.gson.Gson
import com.hypnoweb.hypnowebapp.catalog.Catalog
import io.reactivex.Single
import java.io.InputStreamReader
import javax.inject.Inject

class CatalogNetworkDataSource @Inject constructor(private val catalogApi: CatalogApi,
                                                   private val catalogNetworkMapper: CatalogNetworkMapper,
                                                   context: Context) {

    val records: CatalogNetwork

    init {
        val gson = Gson()
        records = gson.fromJson(InputStreamReader(context.getAssets().open("records.txt"), Charsets.UTF_8), CatalogNetwork::class.java)
    }

    val catalog: Single<Catalog>
        get() = Single.just(records)
                .flatMap { catalogNetwork -> catalogNetworkMapper.fromNetworkCatalog(catalogNetwork) }
}